pkg/splunk/enterprise/searchheadcluster.go

```diff
@@ -211,6 +211,16 @@ func ApplySearchHeadCluster(ctx context.Context, client splcommon.ControllerClie
 		cr.Status.AdminPasswordChangedSecrets = make(map[string]bool)
 		cr.Status.NamespaceSecretResourceVersion = namespaceScopedSecret.ObjectMeta.ResourceVersion
 
+		// Add a splunk operator telemetry app
+		if !cr.Status.TelAppInstalled && !cr.Spec.Mock {
```

Comment:
I may be wrong, but i thought we removed all the references to mock?

Reply:
The pod exec commands via the operator are avoided for unit test using [MockPodExecClient](https://github.com/splunk/splunk-operator/blob/049d00f5eb64c37db4610922e5890021b0fa42d9/pkg/splunk/test/util.go#L61). But the problem here is that unit tests such as **TestClusterMasterWithReadyState**, **TestStandaloneWithReadyState** are directly calling ApplyClusterManager which either takes either the common controller client or [mock test client](https://github.com/splunk/splunk-operator/blob/049d00f5eb64c37db4610922e5890021b0fa42d9/pkg/splunk/test/controller.go#L224) as an argument but won't take MockPodExecClient as an argument. Should we implement mock test client as PodExecClientImpl?

Reply:
I think @vivekr-splunk is using a new framework, to avoid this?

Reply:
That's right @sgontla , but that framework doesn't work here due to the call stack originating from **ApplyClusterManager** which cannot take **MockPodExecClient** as an argument. @vivekr-splunk Any thoughts here?

Reply:
Removed the use of mock and added UT's.